src/app/passengers/components/smart/trip-planner/trip-planner.component.ts

```diff
@@ -167,12 +171,61 @@ export class TripPlannerComponent implements OnInit {
     this.store.dispatch(new MapActions.AddLocation(map));
   }
 
+  getTripEstimate(event){
+    var res;
+
+    const headers = new HttpHeaders({
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'Authorization': 'Basic ' + btoa('front-end:front-end')
+    });
+
+    const data = "grant_type=password&scope=webclient&username=passenger&password=password";
+
+    const options = {
+      headers,
+      withCredentials: true
+    }; 
+
+   this.http.post('http://localhost:32775/auth/oauth/token', data, options).subscribe(res => {
+     if(res){
+       localStorage.setItem("accessToken", JSON.parse(JSON.stringify(res)).access_token)
```

Comment:
`JSON.parse(JSON.stringify(res))` seems like a workaround, do you mind documenting why?

Reply:
this has been corrected to a more simple way. instead of stringifying it and then json parsing we changed it to do as type of any.